pkg/cri/config/config.go

```diff
@@ -46,21 +45,13 @@ type Runtime struct {
 	// When specified, containerd will ignore runtime name field when resolving shim location.
 	// Path must be abs.
 	Path string `toml:"runtime_path" json:"runtimePath"`
-	// Engine is the name of the runtime engine used by containerd.
-	// This only works for runtime type "io.containerd.runtime.v1.linux".
-	// DEPRECATED: use Options instead. Remove when shim v1 is deprecated.
-	Engine string `toml:"runtime_engine" json:"runtimeEngine"`
```

Comment:
Can we error out (or just print a warning) when removed properties are set?

Reply:
There is no easy way to do that (like "detect unmapped TOML fields").
Except to keep these fields and check if those are not empty after unmarshaling, which is probably not what we want.

Reply:
> There is no easy way to do that (like "detect unmapped TOML fields").

I guess we can use https://pkg.go.dev/github.com/pelletier/go-toml/v2#Decoder.DisallowUnknownFields , but it can be another PR

Reply:
Opened https://github.com/containerd/containerd/issues/8270 for further discussion